### PR TITLE
HPCC-26144 LogicalFile details missing actions

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/forms/CopyFile.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    const titleId = useId("title");
+
+    const closeForm = () => {
+        setShowForm(false);
+        reset(defaultValues);
+    };
+
+    const dragOptions: IDragOptions = {
+        moveMenuItemText: "Move",
+        closeMenuItemText: "Close",
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/CopyFile.tsx:80]: Should use nlsHPCC for translations.

**Review by GordonSmith — CHANGES_REQUESTED** [review]
I stopped reviewing after the first form as I suspect my comments there are relevant to all the forms?

File: esp/src/src-react/components/forms/CopyFile.tsx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+export const CopyFile: React.FunctionComponent<CopyFileProps> = ({
+    cluster,
+    logicalFile,
+    showForm,
+    setShowForm
+}) => {
+
+    const [file] = useFile(cluster, logicalFile);
+
+    const { handleSubmit, control, reset } = useForm<CopyFileFormValues>({ defaultValues });
```
[the remark this answers is not in the document]
Comment (jeclrsg) [reply esp/src/src-react/components/forms/CopyFile.tsx:54]: No, defaultValues is also a key for the config object passed into useForm

File: esp/src/src-react/components/forms/CopyFile.tsx [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+                            rules={{
+                                min: {
+                                    value: 1,
+                                    message: `${nlsHPCC.ValidationErrorNumberLess} 1`
+                                }
+                            }}
+                        /></td>
+                    </tr></tbody>
+                </table>
+            </Stack>
```
[the remark this answers is not in the document]
Comment (jeclrsg) [reply esp/src/src-react/components/forms/CopyFile.tsx:273]: I think so. The validation, resetting of default values, and submission mechanism this module provides seem pretty nice.

File: esp/src/src-react/components/forms/CopyFile.tsx [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    >
+        <div className={componentStyles.header}>
+            <span id={titleId}>{nlsHPCC.Copy}</span>
+            <IconButton
+                styles={iconButtonStyles}
+                iconProps={cancelIcon}
+                ariaLabel="Close popup modal"
+                onClick={closeForm}
+            />
+        </div>
```
[the remark this answers is not in the document]
Comment (jeclrsg) [reply esp/src/src-react/components/forms/CopyFile.tsx:160]: I'm not sure. I would defer to your opinion, but I think of dialog type of components as a replacement for simpler things, sort of like JS alert / confirm. Whereas modals are for things that might be more complicated, like forms and such.

**Review by GordonSmith — DISMISSED** [review]

**Comment by richardkchapman** [comment]
Needs rebasing for a clean merge

File: esp/src/src-react/components/FileDetails.tsx [hunk 5]
```diff
@@ -47,7 +59,14 @@ export const FileDetails: React.FunctionComponent<FileDetailsProps> = ({
         setProtected(_protected || isProtected);
         setRestricted(restricted || file?.IsRestricted);
 
-    }, [_protected, description, file?.Description, file?.IsRestricted, isProtected, restricted]);
+        if (file?.DFUFilePartsOnClusters?.DFUFilePartsOnCluster?.length > 0) {
+            const part = file?.DFUFilePartsOnClusters?.DFUFilePartsOnCluster[0];
+            setCanReplicateFlag(part.CanReplicate);
+            setReplicateFlag(part.Replicate);
+        }
```
Comment (GordonSmith) [inline esp/src/src-react/components/FileDetails.tsx:66]: Should this check all the parts? - maybe not if their state is always the same?

**Review by GordonSmith — DISMISSED** [review]

**Comment by richardkchapman** [comment]
Still needs rebasing...

**Review by GordonSmith — DISMISSED** [review]

**Comment by GordonSmith** [comment]
@jeclrsg please rebase

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**